glean-core/android/src/main/java/mozilla/telemetry/glean/rust/LibGleanFFI.kt

```diff
@@ -65,11 +66,30 @@ internal fun Pointer.getRustString(): String {
 @Suppress("TooManyFunctions")
 internal interface LibGleanFFI : Library {
     companion object {
-        private val JNA_LIBRARY_NAME = "glean_ffi"
-
         internal var INSTANCE: LibGleanFFI = try {
-            val lib = Native.load(JNA_LIBRARY_NAME, LibGleanFFI::class.java) as LibGleanFFI
-            lib.glean_enable_logging()
+            var libName = BuildConfig.JNA_LIBRARY_NAME
+            val isMegazorded = (libName == "megazord")
+            // If we're part of the application-services megazord, we may need to dynamically
+            // look up the library name in system properties, to work with custom megazords.
+            // Ref https://github.com/mozilla/application-services/blob/main/docs/design/megazords.md
+            if (isMegazorded) {
+                val mzLibName = System.getProperty("mozilla.appservices.megazord.library")
+                if (mzLibName != null) {
+                    val mzLibVersion = System.getProperty("mozilla.appservices.megazord.version")!!
+                    if (mzLibVersion != BuildConfig.LIBRARY_VERSION) {
```

Comment:
Huh, I guess we never really bothered with that [build variable](https://github.com/mozilla/glean/blob/dc60dc2d370fe0107f959bc74de10c3e0ddfce98/glean-core/android/build.gradle#L42)